packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1574,6 +1574,32 @@ void main() {
     expect(RawKeyboard.instance.keyEventHandler, same(rawKeyEventHandler));
   });
 
+  testWidgets('Incorrect use of ParentDataWidget shows a useful error message', (WidgetTester tester) async {
```

Comment:
Do we need an additional test for the Stack/Positioned case?

Reply:
So there are already tests for this which I've found in `parent_test_data.dart`, which already covers the case that I've tested in this PR. I've edited the tests in `parent_test_data` to work with the new error messages and I now think the test I have made is redundant, and actually would not have failed before I made the change in `framework.dart`, so I think I need a better test, what do you think?

I can't currently think of a way to test release-mode code though - which is what this PR addresses, so I'm going to look into this.

Reply:
The original code I modified relied on code in an `assert` to control the flow, and hence running into nasty exceptions only in release mode. I have made a contrived example below demonstrating the issue in the original code:

```dart
num returnDivisionCalculation() {
  num numerator = 1;
  num denominator = 0;
  bool attemptCalculation = true;

  assert(() {
    if (denominator == 0) {
      attemptCalculation = false;

      try {
        throw "Specific division error";
      } catch {
        // log the error and continue
      }
    }
  }());

  if (attemptCalculation) { // Always true in release mode.
    return numerator / denominator;
  }

  return 0;
}
```

In debug mode, the code in `assert` will be run, causing safe and controlled execution of the code, but in release mode an unexpected exception will be thrown.

I don't think it's possible to write a failing test that will succeed after my change and I will ask for an exemption on Discord. I have edited the tests to update the new error messages for misuse of these widgets (of which I am happy to revert, just thought I might improve them while here)